src/iris/api.py

```diff
@@ -729,6 +740,57 @@ def is_valid_tracking_settings(t, k, tpl):
     return True, None
 
 
+def gen_tag_where_subquery(connection, id_field, tag_table, resource_id, kwargs):
+    '''
+        return a subquery to be used in a where clause for filtering based on tags
+    '''
+    tag_prefix = 'tag_'
+    where = []
+    conditions = []
+    joins = []
+    for i, (key, values) in enumerate(kwargs.items()):
+        if not isinstance(key, str):
+            continue
+        # get tag name and any operation associated with it
+        col, _, op = key.partition('__')
+        # only process tag fields
+        if not col.startswith(tag_prefix):
+            continue
+        if op not in operators:
+            raise HTTPBadRequest('invalid filter operator %s' % op)
+
+        name = col[len(tag_prefix):]
+        # ensure proper typing and escape values
+        if isinstance(values, str):
+            values = values.split(',')
+        values = [connection.escape(x) for x in values if isinstance(x, str)]
+        escaped_name = connection.escape(name)
+
+        # Generate conditions for each tag name and its values
+        condition_bits = []
+        for value in values:
+            value_op = operators[op] % ('value', value)
+            condition_bits.append(f"(t{i + 1}.name = {escaped_name} AND t{i + 1}.{value_op})")
+        # a single tag name can have a list of values associated with it, evaluate those as OR
+        condition = " OR ".join(x for x in condition_bits)
+        conditions.append(f"({condition})")
+
+        # Create self-join for each tag name and its values this is necessary for allow matching to multiple tag name:value combiations
+        join_condition = f"t{i + 1}.{id_field} = t0.{id_field}"
+        joins.append(f"JOIN {tag_table} AS t{i + 1} ON {join_condition}")
+
+    if len(conditions) == 0:
+        return ""
+
+    # Combine all conditions using AND to ensure they are all satisfied
+    combined_conditions = " AND ".join(conditions)
+    # Combine all joins
+    combined_joins = " ".join(joins)
+    # SQL query to retrieve distinct resource ids based on the conditions and self-joins
+    sql_query = f"EXISTS (SELECT t0.{id_field} FROM {tag_table} AS t0 {combined_joins} WHERE {combined_conditions} AND t0.{id_field} = {resource_id})"
```

Comment:
With a `exists(select` query, we don't need to include an actual field name in the SELECT clause as it gets ignored 

Eg we can just do `EXISTS (SELECT 1 FROM `

Reply:
changed